.github/workflows/auto-update-collector-version

```diff
@@ -0,0 +1,32 @@
+name: Auto-update Collector Version
```

Comment:
I wouldn't call this auto-update, since the workflow doesn't, for example, both check and then automatically commit the new version to `main`.

This script both checks the version and submits a PR if necessary. That's a bit of a mouthful, so maybe just name this "Check collector version".

```suggestion
name: Check collector version
```

If you agree, then the file name should be updated too.

Reply:
agreed